src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -644,6 +649,9 @@ CreateHashDistributedTable(Oid relationId, char *distributionColumnName,
 		}
 		else
 		{
+			/* to allow concurrent run with another create_distributed_table */
+			heap_close(pgDistColocation, ExclusiveLock);
+			heap_open(DistColocationRelationId(), AccessShareLock);
```

Comment:
I don't think we need to get an `AccessShareLock` on `pg_dist_colocation` here, this would only protect us against the extension getting dropped, but there are various other protections against that.